app/components/application/auth-onboarding/sign-in.js

```diff
@@ -12,31 +12,52 @@ export default Component.extend({
 
   facebook: service(),
   notify: service(),
+  aozoraConflicts: service(),
   router: service('-routing'),
   authentication: taskGroup().drop(),
 
   login: task(function* () {
     const { identification, password } = getProperties(this, 'identification', 'password');
-    yield get(this, 'session')
-      .authenticateWithOAuth2(identification, password)
-      .then(() => invokeAction(this, 'close'))
-      .catch(err => get(this, 'notify').error(errorMessages(err)));
+    try {
+      yield get(this, 'session').authenticateWithOAuth2(identification, password);
+      yield get(this, 'gotoNext').perform();
+    } catch (err) {
+      get(this, 'notify').error(errorMessages(err));
+    }
   }).group('authentication'),
 
   loginWithFacebook: task(function* () {
-    yield get(this, 'session').authenticateWithFacebook()
-      .then(() => invokeAction(this, 'close'))
-      .catch((error) => {
-        // Facebook succeeded but Kitsu failed (no-account)
-        if (error.error === 'invalid_grant') {
-          get(this, 'facebook').getUserData().then((response) => {
-            const data = { ...response, name: underscore(get(response, 'name')) };
-            invokeAction(this, 'changeComponent', 'sign-up', data);
-          }).catch(err => get(this, 'notify').error(errorMessages(err)));
+    try {
+      yield get(this, 'session').authenticateWithFacebook()
+      yield get(this, 'gotoNext').perform();
+    } catch (error) {
+      // Facebook succeeded but Kitsu failed (no-account)
+      if (error.error === 'invalid_grant') {
+        try {
+          const response = get(this, 'facebook').getUserData();
+          const data = { ...response, name: underscore(get(response, 'name')) };
+          invokeAction(this, 'changeComponent', 'sign-up', data);
+        } catch (err) {
+          get(this, 'notify').error(errorMessages(err));
         }
-      });
+      }
+    }
   }).group('authentication'),
 
+  gotoNext: task(function* () {
+    const user = yield get(this, 'session').getCurrentUser();
+    if (user) {
+      const conflicts = yield get(this, 'aozoraConflicts').list();
+      if (conflicts.length > 1) {
+        invokeAction(this, 'changeComponent', 'aozora-conflict');
+      } else {
+        invokeAction(this, 'changeComponent', 'aozora-account-details');
+      }
+    } else {
+      invokeAction(this, 'close');
+    }
+  }),
+
```

Comment:
Isn't this going to force the user to the `aozora-account-details` component after signing in and no conflicts?

Reply:
Oh shit you're right, yeah